File: exercise/12/solution.cpp
class Solution {
public:
    string intToRoman(int num) {
        string roman = "IVXLCDM";
        string result;
        int tnum = num, pos = 0;
        while (tnum) {
            string delta = "";
            int digit = tnum % 10;
            tnum /= 10;
            switch (digit) {
                case 9:
                    delta.push_back(roman[pos]);
                    delta.push_back(roman[pos+2]);
                    break;
                case 8:
                    delta.push_back(roman[pos]);
                case 7:
                    delta.push_back(roman[pos]);
                case 6:
                    delta.push_back(roman[pos]);
                case 5:
                    delta = roman[pos+1] + delta;
                    break;
                case 4:
                    delta.push_back(roman[pos]);
                    delta.push_back(roman[pos+1]);
                    break;
                case 3:
                    delta.push_back(roman[pos]);
                case 2:
                    delta.push_back(roman[pos]);
                case 1:
                    delta.push_back(roman[pos]);
            }
            result = delta + result;
            pos += 2;
        }
        return result;
    }
};

File: exercise/12/solution.py
class Solution:
    def intToRoman(self, num: int) -> str:
        roman = "IVXLCDM"
        result = ""
        i = 0
        while num > 0:
            digit = num % 10
            num //= 10
            now = ""
            if digit == 9:
                now = roman[i] + roman[i+2]
            elif digit >= 5:
                now = roman[i+1]
                for j in range(5, digit):
                    now += roman[i]
            elif digit == 4:
                now = roman[i] + roman[i+1]
            else:
                for j in range(0, digit):
                    now += roman[i]
            result = now + result
            i += 2
        return result
